src/unix.rs

```diff
@@ -228,7 +232,8 @@ fn make_os_str(slice: &[u8]) -> Result<EitherOsStr, Error> {
             panic!("Path to file cannot contain nul-byte in the middle");
         }
         if last == 0 {
-            let str = unsafe { OsStr::from_slice(transmute(slice)) };
+            let str =
```

Comment:
Notice how in this block, the Borrow slice _included_ the null byte. Now that will be intentional.